test/types/objectSearchCreate.test.tsx

```diff
@@ -103,6 +106,32 @@ describe('objectSearchCreate', () => {
     expect(onSave).toHaveBeenCalledWith(props.model);
   });
 
+  it('Clears existing record on add new', async () => {
+    const { field, onSave, searchTerm, result, props } = getDefaults({
+        model: { law_firm: fakeLawFirm() },
+      })
+      , tester = await getTester(props)
+      ;
+
+    await objectSearchFor(tester, field, result, searchTerm);
+    await selectAddNew(tester);
+
+    // Will not submit until required sub-form filled out
+    await tester.refresh();
+    tester.submit();
+    expect(onSave).not.toHaveBeenCalled();
+    expect(tester.text()).toContain('Required');
+
+    // Will clear errors when fixing invalid field
+    tester.changeInput('input[id="law_firm.name"]', searchTerm);
+    tester.submit();
+    expect(onSave).toHaveBeenCalledWith({ law_firm: { name: searchTerm, amount_owed: null }});
+
+    tester.click('.ant-input-search-create-btn-back');
+    tester.submit();
+    expect(onSave).toHaveBeenCalledWith(props.model);
```

Comment:
It seems unintuitive for this to be filled upon clicking back. What's more intuitive is that each state change (to add new, to search create) zeroes out the submit data